tests/aggregation/tests.py

```diff
@@ -2359,3 +2390,256 @@ def test_aggregate_combined_queries(self):
             )
         )
         self.assertEqual(qs.count(), 3)
+
+
+class StringAggTestModel(Model):
+    char_field = CharField(max_length=30, blank=True)
+    text_field = TextField(blank=True)
+    json_field = JSONField(null=True)
+
+
+class StatTestModel(Model):
+    int1 = IntegerField()
+    int2 = IntegerField()
+    related_field = ForeignKey(StringAggTestModel, SET_NULL, null=True)
```

Comment:
Try reuse the existing models defined in `aggregation/models.py` instead. We can't add new tables/models for each new features otherwise the suite slowly gets slower.

Reply:
These tests are taken directly from the PostgreSQL specific of these tests. My thoughts were

1. We are deprecating that class in 6.0, so we will delete all of the StringAgg tests from that module at that time
2. Those tests exist for a reason, even if I am not familiar with them, so I didn't want to drop any and cause a regression

This is why I have some tests that may seem very specific, like the subqueries, and why I went about defining these models. It kept me from re-implementing the same behaviors in a way that might cause a regression. However, I am open to using the models already defined in `aggregation/models.py`, I'll just need to sit with the data a bit to come up with the right expectations. I think it might be best to keep the tests, like the subqueries, in place to make sure we maintain the coverage. If you think those behaviors are covered in other tests, though, then I'm open to cleaning them up.

Reply:
That makes sense, if they already exist we should definitely just port them over.

> However, I am open to using the models already defined in aggregation/models.py

If you could port them while adapting them to use the existing models that would be awesome.

Reply:
Also, if we're porting them over we should delete the original ones from `postgres_tests` given `postgres.aggregates.StringAgg` will only a shim at this point.

Reply:
I did some work to merge the tests into the existing `AggregateTestCase`, keeping some of the more esoteric ones specific to the `StringAgg` behaviors but making them work with the existing data/models. I had to add a JSON field to the existing models to that end, but otherwise, I think it works pretty well.